content/post/binding-class-methods-react.md

```diff
@@ -0,0 +1,77 @@
++++
+date = "2016-06-09T13:28:29-07:00"
+draft = false
+title = "Binding es6 class Methods React"
+categories = ["react", "javascript"]
+tags = ["tips", "javascript", "react", "es6"]
+
++++
+
+The choice between using the es6 `class` method for declaring component in
+React seems to be a decided convention within the React community. The
+only issue with it the need to bind the context of `this` everytime you
+create a function. This way you have 
+
+The [Netlify login](https://app.netlify.com/login) form is an easy enough component to look at:
+
+```js
+class LoginPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {loginError: null};
+  }
+
+  onLogin(provider) {
+    return (event) => {
+      ...
+    };
+  }
+
+  render() {
+    const login = () => this.onLogin.bind;
+    return <LoginForm handleLogin={login} loginError={this.state.loginError}/>;
+  }
+}
+```
+Everytime you need to use a class function you need to either bind using
+a function `() => this.onLogin` or binding directly
+`this.onLogin.bind(this)`. The ladder won't pass the default settings
+for [eslint](http://eslint.org/docs/rules/no-extra-bind). The first
+option seems less hacky, but you have to redeclare a new function. 
+
+A new way to handling the the binding of class methods do this has arisen and it's actually pretty nice. Just add the `bind(this)` to the class's constructor.
+
+```js
+constructor(props) {
+  super(props);
+  ...
+  this.onLogin = this.onLogin.bind(this);
+}
+
+```
+
+Now you can call just call `this.onLogin` directly, which save a bit
+processing power by not need to create a new function:
```

Comment:
I guess what I was getting at is one is fine, but most components have multiple function calls that need to bind() this to. It adds up when minified, but regardless, going to go with a different conclusion.